src/Microsoft.AspNetCore.Blazor.Browser/Hosting/WebAssemblyHost.cs

```diff
@@ -97,5 +101,27 @@ public void Dispose()
         {
             (Services as IDisposable)?.Dispose();
         }
+
+        private static void SetBrowserHttpMessageHandlerAsDefault()
+        {
+            // Within the Mono WebAssembly BCL, this is a special private static field
+            // that can be assigned to override the default handler
+            const string getHttpMessageHandlerFieldName = "GetHttpMessageHandler";
+            var getHttpMessageHandlerField = typeof(HttpClient).GetField(
+                getHttpMessageHandlerFieldName,
+                BindingFlags.Static | BindingFlags.NonPublic);
+
+            // getHttpMessageHandlerField will be null in tests, but nonnull when actually
+            // running under Mono WebAssembly
+            if (getHttpMessageHandlerField != null)
+            {
+                // Just in case you're not actually using HttpClient, defer the construction
+                // of the BrowserHttpMessageHandler
+                var handlerSingleton = new Lazy<HttpMessageHandler>(
+                    () => new BrowserHttpMessageHandler());
+                Func<HttpMessageHandler> handlerFactory = () => handlerSingleton.Value;
+                getHttpMessageHandlerField.SetValue(null, handlerFactory);
+            }
```

Comment:
Does it make sense for us to make a lot of noise if this fails? I'm not sure we have a good way to report it other than logging to the console, but it seems worth talking about.